snomed/com.b2international.snowowl.snomed.api.rest.tests/src/com/b2international/snowowl/snomed/api/rest/SnomedCompareRestRequestTest.java

```diff
@@ -0,0 +1,231 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import com.jayway.restassured.http.ContentType;
+import com.jayway.restassured.response.Response;
+/**
+ * @since 7.0
+ */
+public class SnomedCompareRestRequestTest extends AbstractSnomedApiTest {
+	
+	private IBranchPath parentBranch;
+	private IEventBus bus;
+
+	private String repositoryUuid;
+	
+	private ImmutableMap<String, Object> createCompareRequest(IBranchPath base, IBranchPath compareBranch) {
+		return ImmutableMap.<String, Object>builder()
+				.put("base", base.toString())
+				.put("compareBranch", compareBranch.toString())
+				.build();
+	}
+	
+	private ImmutableMap<String, Object> createCompareRequest(String base, String compareBranch) {
+		return ImmutableMap.<String, Object>builder()
+				.put("base", base)
+				.put("compareBranch", compareBranch)
+				.build();
+	}
+	
+	@Before
+	public void setup() {
+		parentBranch = BranchPathUtils.createPath("MAIN/SnomedCompareRestRequestTest");
+		bus = ApplicationContext.getServiceForClass(IEventBus.class);
+		repositoryUuid = SnomedDatastoreActivator.REPOSITORY_UUID;
+	}
+	
+	@Test
+	public void testCompareOfSameBranch() throws IOException {
+		final ImmutableMap<String, Object> compareRequest = createCompareRequest(branchPath, branchPath);
+		
+		final InputStream responseInputStream = givenAuthenticatedRequest(SCT_API)
+			.contentType(ContentType.JSON)
+			.accept(ContentType.JSON)
+			.body(compareRequest)
+			.post("/compare")
+			.asInputStream();
+		final ObjectMapper mapper = new ObjectMapper();
+		final CompareResult responseCompareResult = mapper.readValue(responseInputStream, CompareResult.class);
+		assertThat(responseCompareResult.getBaseBranch().equals(branchPath.toString()));
+		assertThat(responseCompareResult.getCompareBranch().equals(branchPath.toString()));
+		assertThat(responseCompareResult.getChangedComponents().isEmpty());
+		assertThat(responseCompareResult.getDeletedComponents().isEmpty());
+		assertThat(responseCompareResult.getNewComponents().isEmpty());
+	}
+	
+	@Test
+	public void testCompareOfNonExistantBranch() {
+		final ImmutableMap<String, Object> compareRequest = createCompareRequest(branchPath, BranchPathUtils.createPath("MAIN/Branch/does/not/exist"));
+		
+		final Response compareResponse = givenAuthenticatedRequest(SCT_API)
+				.contentType(ContentType.JSON)
+				.accept(ContentType.JSON)
+				.body(compareRequest)
+				.post("/compare")
+				.thenReturn();
+		
+		 final int statusCode = compareResponse.statusCode();
+		 assertEquals(404, statusCode);
+	}
+	
+	@Test
+	public void testCompareWithChangedNewComponents() throws IOException {
+		final String newConceptId = createNewConcept(branchPath);
+		final ImmutableMap<String, Object> compareRequest = createCompareRequest(parentBranch, branchPath);
+		
+		final InputStream responseInputStream = givenAuthenticatedRequest(SCT_API)
```

Comment:
Extract common `compare` functionality to helper methods to make tests more readable.